### PR TITLE
Pass full configuration registry object to add-ons on bootstrap time + component registry API refactor

### PR DESCRIPTION
**Pull request by sneridagh**
AKA named component lookup in Volto, FTW

### CODE REVIEW

File: src/registry.test.js [hunk 1]
```diff
@@ -3,25 +3,41 @@ import config from './registry';
...
+  it('resolves unexistent component', () => {
+    expect(config.getComponent('Toolbar.Doh').component).toEqual(undefined);
+    expect(config.getComponent('Toolbar.Doh')).toEqual({});
   });
-  it('register component by name', () => {
-    config.register('Toolbar.Bar', { component: 'this is a Bar component' });
-    expect(config.resolve('Toolbar.Bar').component).toEqual(
+  it('registers a component by name', () => {
+    config.registerComponent('Toolbar.Bar', {
+      component: {
```
Comment (sneridagh) [inline src/registry.test.js:36]: I don't like this... too ugly :(

File: src/registry.js [hunk 2]
```diff
@@ -88,12 +88,21 @@ class Config {
     this._data.components = components;
   }
 
-  resolve(component) {
-    return this._data.components[component] || {};
+  getComponent(name, options = {}) {
```
Comment (sneridagh) [inline src/registry.js:91]: Should we go to an all keywords arguments?

File: src/registry.js [hunk 3]
```diff
@@ -88,12 +88,21 @@ class Config {
...
+    const { context = '' } = options;
+    const componentName = `${name}${context ? `|${context}` : ''}`;
```
Comment (sneridagh) [inline src/registry.js:93]: I went for a string based composition for name... We could go for the key based one as well.

File: src/registry.test.js (hunk 1 above)
Comment (tiberiuichim) [reply src/registry.test.js:36]:
```
registerComponent({name: 'Toolbar', component: Toolbar, context: 'Bar'})
```
or:
```
registerComponent('Toolbar', Toolbar, 'Bar')
```
or:
```
registerComponent({name: 'Toolbar', component: Toolbar, dependencies: ['Bar'], options: {something: 'else'}})
```
I think I like the last?

Comment (tiberiuichim) [reply src/registry.test.js:36]: We also use `dependencies` in the Pluggables implementation.

**Review by tiberiuichim — APPROVED** [review]

File: src/registry.test.js (hunk 1 above)
Comment (sneridagh) [reply src/registry.test.js:38]: Done

**Pull request merged**